ts/components/LabelledItem/index.tsx

```diff
@@ -30,43 +36,123 @@ const styles = StyleSheet.create({
   icon?: string | ImageSourcePropType;
+  iconColor?: string;
   iconPosition?: "left" | "right";
   iconStyle?: StyleType;
-  iconColor?: string;
+  inputMaskProps?: TextInputMaskProps &
+    React.ComponentPropsWithRef<typeof TextInputMask>;
+  inputProps?: TextInputAdditionalProps;
   isValid?: boolean;
-  focusBorderColor?: string;
-  description?: string;
-  accessibilityLabel?: string;
-  accessibilityHint?: string;
-  accessibilityLabelIcon?: string;
+  label?: string;
   onPress?: () => void;
-  hasNavigationEvents?: boolean;
 }>;
 
-export type Props = WithTestID<CommonProp> & InputProps;
+export type Props = WithTestID<CommonProp>;
+
+function getColorsByProps({
+  isDisabledTextInput,
+  hasFocus,
+  isEmpty,
+  isValid
+}: {
+  isDisabledTextInput: boolean;
+  hasFocus: boolean;
+  isEmpty: boolean;
+  isValid?: boolean;
+}) {
+  const borderColor = isDisabledTextInput
+    ? IOColors.greyLight
+    : hasFocus && isEmpty
+    ? variables.itemBorderDefaultColor
+    : undefined;
+
+  // weird type needed to be accepted as H5 prop
+  const descriptionColor:
+    | "bluegreyLight"
+    | "red"
+    | "bluegreyDark" = isDisabledTextInput
+    ? "bluegreyLight"
+    : isValid === false
+    ? "red"
+    : "bluegreyDark";
+  const iconColor = isDisabledTextInput
+    ? IOColors.bluegreyLight
+    : variables.brandDarkGray;
+  // weird type needed to be accepted as H5 prop
+  const labelColor: "bluegreyLight" | "bluegreyDark" = isDisabledTextInput
+    ? "bluegreyLight"
+    : "bluegreyDark";
 
+  const placeholderTextColor = isDisabledTextInput
+    ? color(IOColors.bluegreyLight).string()
+    : color(variables.brandGray).darken(0.2).string();
+
+  return {
+    borderColor,
+    descriptionColor,
+    iconColor,
+    placeholderTextColor,
+    labelColor
+  };
+}
 export const LabelledItem: React.FC<Props> = ({
   iconPosition = "left",
   ...props
 }: Props) => {
   const [isEmpty, setIsEmpty] = useState(true);
   const [hasFocus, setHasFocus] = useState(false);
 
-  const descriptionColor = props.isValid === false ? "red" : "bluegreyDark";
   const accessibilityLabel = props.accessibilityLabel ?? "";
-  const inputBorderColor =
-    hasFocus && isEmpty
-      ? variables.itemBorderDefaultColor
-      : props.focusBorderColor;
   const isValid = props.isValid === undefined ? false : props.isValid;
   const isNotValid = props.isValid === undefined ? false : !props.isValid;
 
+  const {
+    borderColor,
+    descriptionColor,
+    iconColor,
+    placeholderTextColor,
+    labelColor
+  } = getColorsByProps({
+    isDisabledTextInput: Boolean(props.inputProps && props.inputProps.disabled),
+    hasFocus,
+    isEmpty,
+    isValid: props.isValid
+  });
+
+  const handleOnFocus = () => {
+    setHasFocus(true);
+  };
+
+  const handleOnBlur = () => {
+    setHasFocus(false);
+  };
+
+  /**
+   * check if the input is empty and set the value in the state
+   */
+  const checkInputIsEmpty = (text?: string) => {
```

Comment:
```suggestion
  const checkInputIsEmpty = (text?: string) => setIsEmpty(isStringNullyOrEmpty(text));
```
the same but using an util function